core/src/main/java/org/apache/iceberg/rest/HttpRESTClient.java

```diff
@@ -0,0 +1,288 @@
+ */
+public class HttpRESTClient implements RESTClient {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HttpRESTClient.class);
+
+  private final String uri;
+  private final CloseableHttpClient httpClient;
+  private final ObjectMapper mapper;
+  private final Map<String, String> additionalHeaders;
+  private final Consumer<HttpUriRequest> requestInterceptor;
+
+  private HttpRESTClient(
+      String uri,
+      CloseableHttpClient httpClient,
+      ObjectMapper mapper,
+      Map<String, String> additionalHeaders,
+      Consumer<HttpUriRequest> requestInterceptor) {
+    this.uri = uri;
+    this.httpClient = httpClient != null ? httpClient : HttpClients.createDefault();
+    this.mapper = mapper != null ? mapper : new ObjectMapper();
+    this.additionalHeaders = additionalHeaders != null ? additionalHeaders : ImmutableMap.of();
+    this.requestInterceptor = requestInterceptor;
+  }
+
+  private static Optional<String> extractResponseBodyAsString(CloseableHttpResponse response, Class<?> responseType) {
+    try {
+      if (response.getEntity() == null) {
+        return Optional.empty();
+      }
+      return Optional.ofNullable(EntityUtils.toString(response.getEntity(), "UTF-8"));
+    } catch (IOException | ParseException e) {
+      if (responseType != null) {
+        LOG.warn("Failed to convert HTTP response body to string", e);
+      }
+      return Optional.empty();
+    }
+  }
+
+  // Per the spec, the only currently defined / used "success" responses are 200 and 202.
+  private static boolean isSuccessful(CloseableHttpResponse response) {
+    int code = response.getCode();
+    return code ==  HttpStatus.SC_OK || code == HttpStatus.SC_ACCEPTED;
+  }
+
+  private static ErrorResponse buildDefaultErrorResponse(CloseableHttpResponse response) {
+    String responseReason = response.getReasonPhrase();
+    String message =
+        responseReason != null && !responseReason.isEmpty() ? responseReason :
+            EnglishReasonPhraseCatalog.INSTANCE.getReason(response.getCode(), null /* ignored */);
+    String type = "ClientGeneratedErrorResponse - No error response body could be parsed";
+    return ErrorResponse.builder()
+        .responseCode(response.getCode())
+        .withMessage(message)
+        .withType(type)
+        .build();
+  }
+
+  private static ErrorResponse parseOrCreateErrorResponse(
+      CloseableHttpResponse response, Optional<String> responseBody, Method method, Class<?> responseType) {
+    try {
+      return responseBody.map(ErrorResponseParser::fromJson).orElseGet(() -> buildDefaultErrorResponse(response));
+    } catch (Exception e) {
+      // Only log a warning that we couldn't parse an error response if some response body was expected
+      if (responseType != null) {
+        LOG.warn(
+            "Failed to parse standard error response for {} call expecting {} response. Creating one instead",
+            method, responseType.getSimpleName(), e);
```

Comment:
This is the only place where responseType and method are used, but I'm not sure that these are actually valuable here. This is parsing an error response, so including the expected response type is probably confusing. And the method will be clear from the stack trace. I think you can remove both.

Reply:
Both have been removed and just the exception (and the message) is logged.